usr/src/uts/common/fs/zfs/dbuf.c

```diff
@@ -2940,7 +2916,8 @@ dbuf_rele_and_unlock(dmu_buf_impl_t *db, void *tag)
 				    db->db.db_size, db);
 				mutex_exit(&db->db_mtx);
 
-				if (db->db_caching_status == DB_DBUF_CACHE) {
+				if (db->db_caching_status == DB_DBUF_CACHE &&
+				    !evicting) {
 					dbuf_evict_notify();
```

Comment:
What if pass "evicting" param to the function as a "don't_do_direct_eviction" param so eviction thread will always be awakened ASAP?

Reply:
That makes sense.  I think in practice it isn't a big deal either way, because we go through dbuf_rele_and_unlock(evicting=FALSE) so often (e.g. nearly every call into the ZPL/ZVOL is going to do this).  And when evicting=TRUE, we typically are either already in the dbuf_evict_thread, or we are called from dbuf_rele_unlock(evicting=FALSE), so we checked (and signaled if necessary) just a few nanoseconds ago.

Reply:
Thank you. I have no other questions.